logging/nxscope/nxscope.c

```diff
@@ -0,0 +1,1142 @@
+
+          if (dlen != (sizeof(struct nxscope_set_frame_s) +
+                       NXSCOPE_DIV_LEN - 1))
+            {
+              _err("ERROR: invalid div single dlen = %d\n", dlen);
+              ret = -EINVAL;
+              goto errout;
+            }
+
+          if (set->chan > s->cmninfo.chmax)
+            {
+              ret = -EINVAL;
+              goto errout;
+            }
+
+          /* Write configuration */
+
+          s->chinfo[set->chan].div = data->ch[0].div;
+
+          break;
+        }
+
+      case NXSCOPE_SET_REQ_BULK:
+        {
+          /* Verify request length */
+
+          if (dlen != (sizeof(struct nxscope_set_frame_s) +
+                       NXSCOPE_DIV_LEN * s->cmninfo.chmax - 1))
+            {
+              _err("ERROR: invalid div bulk dlen = %d\n", dlen);
+              ret = -EINVAL;
+               goto errout;
+            }
+
+          /* Write configuration */
+
+          for (i = 0; i < s->cmninfo.chmax; i += 1)
+            {
+              s->chinfo[i].div = data->ch[i].div;
+            }
+
+          break;
+        }
+
+      case NXSCOPE_SET_REQ_ALL:
+        {
+          /* Verify request length */
+
+          if (dlen != (sizeof(struct nxscope_set_frame_s) +
+                       NXSCOPE_DIV_LEN - 1))
+            {
+              _err("ERROR: invalid div all dlen = %d\n", dlen);
+              ret = -EINVAL;
+              goto errout;
+            }
+
+          /* Write configuration */
+
+          for (i = 0; i < s->cmninfo.chmax; i += 1)
+            {
+              s->chinfo[i].div = data->ch[0].div;
+            }
+
+          break;
+        }
+
+      default:
+        {
+          _err("ERROR: invalid set->req=%d\n", set->req);
+          ret = -EINVAL;
+          goto errout;
+        }
+    }
+
+errout:
+  return ret;
+}
+#endif
+
+/****************************************************************************
+ * Name: nxscope_start_set
+ *
+ * NOTE: This function assumes that we have exclusive access to the nxscope
+ *       instance
+ *
+ ****************************************************************************/
+
+static int nxscope_start_set(FAR struct nxscope_s *s, bool start)
+{
+  int ret = OK;
+
+  s->start = start;
+
+  /* User specific callback */
+
+  if (s->callbacks != NULL && s->callbacks->start != NULL)
+    {
+      ret = s->callbacks->start(s->callbacks->start_priv, start);
+      if (ret < 0)
+        {
+          _err("ERROR: s->callbacks->start failed %d\n", ret);
+        }
+    }
+
+  return ret;
+}
+
+/****************************************************************************
+ * Name: nxscope_start_req
+ *
+ * NOTE: This function assumes that we have exclusive access to the nxscope
+ *       instance
+ *
+ ****************************************************************************/
+
+static int nxscope_start_req(FAR struct nxscope_s *s,
+                             FAR struct nxscope_start_data_s *data)
+{
+  int ret = OK;
```

Comment:
```suggestion
  int ret;
```
or init `ret` to `-EINVAL` and rework condition to
```
int ret = -EINVAL;
...
if (data->start == 0 || data->start == 1)
  {
    _info("data->start=%d\n", data->start);
    ret = nxscope_start_set(s, data->start);
  }
```

Reply:
That looks much better